pwa-devdocs/src/tutorials/create-taglist-component/index.md

```diff
@@ -0,0 +1,464 @@
+This module is the default module exported by the `index.js` file.
+
+Open the `tagList.js` file and add the following content:
+
+```jsx
+import React from 'react';
+
+import { arrayOf, string } from 'prop-types';
+
+// Import the previously defined Tag component
+import Tag from './tag';
+
+// This is the main tag list module
+const TagList = props => {
+
+    // Destructure the props object into variables
+    const { tagArray } = props;
+
+    // Convert the array of tag strings into a list of Tag components
+    const tagList = tagArray.map(value => {
+        return <Tag key={value} value={value} />;
+    })
+
+    // Returns the list of Tag components inside a ul container
+    return (<ul>{tagList}</ul>);
+}
+
+TagList.propTypes = {
+    // tagArray is expected to be an array of strings
+    tagArray: arrayOf(string)
+}
+
+export default TagList;
+```
+
+## Step 4. Expose the module's public API
+
+Open the component's `index.js` file and add the following content:
+
+```js
+export { default } from './tagList';
+```
+
+Exporting modules from your component's `index.js` file is a commond standard in a React project.
+It allows a component to import this component using its directory name.
+
+## Step 5. Add component to the product details page
+
+Find the component that renders the product details page in your storefront project and insert the TagList component into the page.
+
+In the `pwa-studio` project, this component is defined in the following file:
+
+```text
+packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js
+```
+
+At the top of this file, import the TagList component:
+
+```jsx
+import TagList from '../TagList';
+```
+
+If you are using the default Venia ProductFullDetail component you can insert TagList into the **Product Description** section:
+
+```diff
+ <section className={classes.description}>
+     <h2 className={classes.descriptionTitle}>
+         Product Description
+     </h2>
+     <RichText content={productDetails.description} />
++    <TagList tagArray={['Tag1', 'Tag2', 'Tag3']} />
+ </section>
+```
+
+Now, when you visit a product detail page, it will show a list with the hard coded strings.
+
+![TagList with hard coded strings](images/hard-coded-list.png)
+
+## Step 6. Connect to real product data
+
+To make this component useful, it needs to show actual product-specific data.
+Clicking each tag should display other similarly tagged items.
+
+For this tutorial, the TagList component will list the product's categories.
+
+### Edit the GraphQL query
+
+Venia's ProductFullDetail component gets the product data from the GraphQL query defined in:
+
+```text
+packages/venia-ui/lib/queries/getProductDetail.graphql
+```
+
+Edit this file and add the following:
+
+```diff
+ meta_title @include(if: $onServer)
+ # Yes, Products have `meta_keyword` and
+ # everything else has `meta_keywords`.
+ meta_keyword @include(if: $onServer)
+ meta_description @include(if: $onServer)
+ 
++categories {
```

Comment:
`categories` was added to this query for breadcrumbs, and while adding this to the query doesn't error out, it's not giving me the correct data. I would change this to say add `name` and `url_path` to existing category shape.